cpp/include/cuvs/neighbors/cagra_c.h

```diff
@@ -46,15 +46,33 @@ enum cagraGraphBuildAlgo {
  */
 struct cagraIndexParams {
   /** Degree of input graph for pruning. */
-  size_t intermediate_graph_degree = 128;
+  size_t intermediate_graph_degree;
   /** Degree of output graph. */
-  size_t graph_degree = 64;
+  size_t graph_degree;
   /** ANN algorithm to build knn graph. */
-  cagraGraphBuildAlgo build_algo = IVF_PQ;
+  enum cagraGraphBuildAlgo build_algo;
   /** Number of Iterations to run if building with NN_DESCENT */
-  size_t nn_descent_niter = 20;
+  size_t nn_descent_niter;
 };
 
+typedef struct cagraIndexParams* cagraIndexParams_t;
+
+/**
+ * @brief Allocate CAGRA Index params, and populate with default values
+ *
+ * @param[in] params cagraIndexParams_t to allocate
+ * @return cuvsError_t
+ */
+cuvsError_t cagraIndexParamsCreate(cagraIndexParams_t* params);
```

Comment:
Small nitpick, but can we please add cuvs tot he beginning? It makes it a lot easier to know where different functions are coming from when, for example, an application is making calls to other C libraries like cublas and cusparse.